### PR TITLE
Updated yamlProvider to parse time.Duration

### PR DESCRIPTION
**Pull request by anuptalwalkar**
GFM-158

### CODE REVIEW

File: core/config/value.go [hunk 1]
```diff
@@ -421,6 +421,17 @@ func (cv Value) getValueStruct(key string, target interface{}) (interface{}, err
 				continue
 			}
 
+			if fieldType == reflect.TypeOf(time.Duration(0)) {
```
Comment (sectioneight) [inline core/config/value.go]: nit: don't call type of duration in the hot path. define it outside.

File: core/config/yaml_test.go [hunk 2]
```diff
@@ -209,3 +210,22 @@ func TestMapParsingMapWithNonStringKeys(t *testing.T) {
 		assert.Equal(t, "onetwothree", ik.IntKeyMap[123])
 	})
 }
+
+func TestDurationParsing(t *testing.T) {
+	withYamlBytes(t, durationYaml, func(provider Provider) {
+		ds := durationStruct{}
+		err := provider.GetValue("durationStruct").PopulateStruct(&ds)
+		assert.NoError(t, err)
+		expectedDuration, err := time.ParseDuration("30s")
```
Comment (sectioneight) [inline core/config/yaml_test.go]: maybe better to check that it's equal to 30 * time.Second for clarity

Comment (glibsm) [inline core/config/yaml_test.go]: lets also add a few tests for hour and minute durations

File: core/config/value.go (hunk 1 above)
Comment (glibsm) [inline core/config/value.go]: i.e. move `reflect.TypeOf(time.Duration(0))` to a const

File: core/config/yaml_test.go (hunk 2 above)
Comment (anuptalwalkar) [inline core/config/yaml_test.go]: Good idea.

File: core/config/value.go [hunk 3]
```diff
@@ -421,6 +423,17 @@ func (cv Value) getValueStruct(key string, target interface{}) (interface{}, err
 				continue
 			}
 
+			if fieldType == reflect.TypeOf(_zeroTimeDuration) {
```
Comment (sectioneight) [inline core/config/value.go]: Not quite. Call `reflect.TypeOf()` outside of the hot path.

Comment (sectioneight) [inline core/config/value.go]: Yea. Reflect are the expensive calls. It's a micro-optimization, but this is some perf-critical code, so we should aim to reduce expensive calls as much as possible.

Comment (anuptalwalkar) [inline core/config/value.go]: I totally agree with being aware of writing perf critical code in hot paths. #willnothappenagain

Comment (sectioneight) [inline core/config/value.go]: #neverforget

**Pull request merged**